index.ts

```diff
@@ -83,6 +83,102 @@ declare namespace dataSource {
   function create(name: string, params: Array<string>): void
 }
 
+function format(fmt: string, args: string[]): string {
+  let out = ''
+  let argIndex = 0
+  for (let i: i32 = 0; i < fmt.length; i++) {
+    if (i < fmt.length - 1 && fmt.charAt(i) == '{' && fmt.charAt(i + 1) == '}') {
+      if (argIndex >= args.length) {
+        throw Error('Too few arguments for format string: ' + fmt)
+      } else {
+        out += args[argIndex++]
+        i++
+      }
+    } else {
+      out += fmt[i]
+    }
+  }
+  return out
+}
```

Comment:
Minor improvements: 
```ts
function format(fmt: string, args: string[]): string {
  let out = ''
  let argIndex = 0
  for (let i = 0, len = fmt.length - 1; i <= len; i++) {
    if (i < len && fmt.charCodeAt(i) == 0x7B && fmt.charCodeAt(i + 1) == 0x7D) {
       // ... rest the same ...
```
`fmt.charAt(i)/fmt.charAt(i + 1)`  and `'{'/'}'` allocate memory in heap and in static section which could eliminate by using `charCodeAt`

Reply:
Updated, thanks!